docs/docs/guides/tracking/costs.md

```diff
@@ -0,0 +1,163 @@
+# Costs
+
+Costs are an evolving feature of Weave. Currently, the Weave UI calculates costs based on the number of tokens used and the model used, and displays them in the UI, using a static cost file.
+
+In the future, Weave will use a custom cost model, which can be set up by the user. Currently, custom costs can be set up by the user, and fetched by the user in the client, but costs in the UI are still based on token usage, model, and static cost file.
+
+## Adding a custom cost
+
+You can add a custom cost by using the [`add_costs`](/reference/python-sdk/weave/trace/weave.trace.weave_client#method-add_costs) method.
+The two required fields are `prompt_token_cost` and `completion_token_cost`.
+You can also set `effective_date` to a datetime, to make the cost effective at a specific date, this defaults to the current date.
+
+```python
+import weave
+from datetime import datetime
+
+client = weave.init("my_custom_cost_model")
+
+client.add_costs({
+    "your_model_name": {
+        "prompt_token_cost": 0.1,
+        "completion_token_cost": 0.2,
+    }
+})
+
+client.add_costs({
+    "your_model_name": {
+        "prompt_token_cost": 10,
+        "completion_token_cost": 20,
+        # If for example I want to raise the price of the model after a certain date
+        "effective_date": datetime(2025, 4, 22),
+    },
+    "my_special_model_1": {
+        "prompt_token_cost": 0.1,
+        "completion_token_cost": 0.2,
+        "effective_date": datetime(1972, 5, 11),
+    }
+})
+```
+
+## Querying for costs
+
+You can query for costs by using the [`query_costs`](/reference/python-sdk/weave/trace/weave.trace.weave_client#method-query_costs) method.
+There are a few ways to query for costs, you can pass in a singular cost id, or a list of LLM model names.
+
+```python
+import weave
+
+client = weave.init("my_custom_cost_model")
+
+costs = client.query_costs(llm_ids=["your_model_name"])
+
+cost = client.query_costs(costs[0].id)
+```
+
+## Purging a custom cost
+
+You can purge a custom cost by using the `purge_costs` method. You pass in a list of cost ids, and the costs with those ids are purged.
+
+```python
+import weave
+
+client = weave.init("my_custom_cost_model")
+
+costs = client.query_costs(llm_ids=["your_model_name"])
+client.purge_costs([cost.id for cost in costs])
+```
+
+## Setting up a custom cost model
```

Comment:
I would put all of `## Setting up a custom cost model` in a notebook under cookbooks so that you can run it end to end and make sure it is perfect, then link to the generated notebook